lib/kubernetes-deploy/kubernetes_resource/service.rb

```diff
@@ -89,5 +92,14 @@ def related_replica_count
     def external_name_svc?
       @definition["spec"]["type"] == "ExternalName"
     end
+
+    def requires_publishing?
+      @definition["spec"]["type"] == "LoadBalancer"
+    end
+
+    def published?
+      ingresses = @instance_data.dig('status', 'loadBalancer', 'ingress')
+      ingresses && !ingresses.empty?
```

Comment:
As discussed offline - this was changed to `ingresses` because `ingress` is actually a (seemingly improperly named) array, not a single value. `@instance_data.dig('status', 'loadBalancer', 'ingress').present?` should work as well though, I'll update that! 👍